c2/cuda/groth16_cuda.cu

```diff
@@ -280,14 +292,36 @@ RustError::by_value generate_groth16_proofs_c(const Assignment<fr_t> provers[],
         if (caught_exception)
             return;
 
-        split_vectors_l.tail_msms_resize(l_popcount);
-        split_vectors_a.tail_msms_resize(a_popcount);
-        split_vectors_b.tail_msms_resize(b_popcount);
-
-        tail_msm_l_bases.resize(l_popcount);
-        tail_msm_a_bases.resize(a_popcount);
-        tail_msm_b_g1_bases.resize(b_popcount);
-        tail_msm_b_g2_bases.resize(b_popcount);
+        if (l_popcount >= (float)0.5 * points_l.size())
```

Comment:
What's up with `(float)` casts :-) But on a serious note, what is the concern? 0.5 is a floating point number and it will promote the whole expression without any casts...If concern is costs of operations, then float costs as much as double in the context...  BTW, with 0.5 one can just as well do `a >= b/2`:-)

Reply:
Right, thanks! Some of the stuff here are artifacts of an earlier version of the code. I guess this is what happens when I code while feeling under the weather :)